### PR TITLE
feat: MIG create and delete samples

### PR DESCRIPTION
**Pull request by BigBlackWolf**
## Description

Documentation reference: https://cloud.google.com/compute/docs/instance-groups/create-zonal-mig

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/client_library/ingredients/instances/managed_instance_group/create.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    """
+    Send a managed group instance creation request to the Compute Engine API and wait for it to complete.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        group_name: the name for this instance group.
+        size: the size of the instance group.
+        template: the name of the instance template to use for this group. Example:
+            projects/example-project/regions/us-west3-b/instanceTemplates/example-regional-instance-template
```
[the remark this answers is not in the document]
Comment (BigBlackWolf) [reply compute/client_library/ingredients/instances/managed_instance_group/create.py:40]: valid point, fixed

File: compute/client_library/ingredients/instances/managed_instance_group/delete.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+) -> None:
+    """
+    Send a managed group instance deletion request to the Compute Engine API and wait for it to complete.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        group_name: the name for this instance group.
+    Returns:
+        Instance group manager object.
```
[the remark this answers is not in the document]
Comment (BigBlackWolf) [reply compute/client_library/ingredients/instances/managed_instance_group/delete.py:37]: Also good catch

File: compute/client_library/snippets/instances/managed_instance_group/delete.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+) -> None:
+    """
+    Send a managed group instance deletion request to the Compute Engine API and wait for it to complete.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        group_name: the name for this instance group.
+    Returns:
+        Instance group manager object.
```
[the remark this answers is not in the document]
Comment (BigBlackWolf) [reply compute/client_library/snippets/instances/managed_instance_group/delete.py:93]: regenerated

File: compute/client_library/snippets/tests/test_instance_group.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+from ..instances.managed_instance_group.create import create_managed_instance_group
+from ..instances.managed_instance_group.delete import delete_managed_instance_group
+from ..instance_templates.create import create_template
+from ..instance_templates.delete import delete_instance_template
+
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+REGION = "europe-west2"
+ZONE = f"{REGION}-a"
```
[the remark this answers is not in the document]
Comment (BigBlackWolf) [reply compute/client_library/snippets/tests/test_instance_group.py:28]: Nah, with this one I don't agree with you!

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**